### PR TITLE
Migrate to cmake

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

File: agent/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.13)
+cmake_policy(VERSION 3.13)
+
+##############################################################################################################
+
+project(fty-kpi-power-uptime
```
Comment (FrancoisRegisDegott-eaton) [inline agent/CMakeLists.txt]: useless (re)define

File: agent/src/fty_kpi_power_uptime.cc [hunk 2]
```diff
@@ -26,7 +26,8 @@
 @end
 */
 
-#include "fty_kpi_power_uptime_classes.h"
+#include "../include/fty_kpi_power_uptime_library.h"
+
 #define ACTOR_NAME "uptime"
 #define DEFAULT_LOG_CONFIG "/etc/fty/ftylog.cfg"
```
Comment (FrancoisRegisDegott-eaton) [inline agent/src/fty_kpi_power_uptime.cc]: use FTY_COMMON_LOGGING_DEFAULT_CFG instead (from fty_log.h)

File: lib/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.13)
+cmake_policy(VERSION 3.13)
+
+project(fty-kpi-power-uptime
```
Comment (FrancoisRegisDegott-eaton) [inline lib/CMakeLists.txt]: useless (re)define

File: lib/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    DESCRIPTION "Compute Data Center uptime"
+)
+
+##############################################################################################################
+find_package(fty-cmake PATHS ${CMAKE_BINARY_DIR}/fty-cmake)
+##############################################################################################################
+
+etn_target(static ${PROJECT_NAME}-lib
+    SOURCES
+        src/dc.cc
```
Comment (FrancoisRegisDegott-eaton) [inline lib/CMakeLists.txt:9]: *.cc   *.h  make the job

File: lib/src/dc.cc [hunk 5]
```diff
@@ -28,6 +28,13 @@
 
 #include "fty_kpi_power_uptime_classes.h"
 
+/*#include "dc.h"
```
Comment (FrancoisRegisDegott-eaton) [inline lib/src/dc.cc]: to remove?

File: resources/fty-kpi-power-uptime.service.in [hunk 6]
```diff
@@ -0,0 +1,32 @@
+[Unit]
+Description=42ity kpi-power-uptime service
```
Comment (FrancoisRegisDegott-eaton) [inline resources/fty-kpi-power-uptime.service.in]: @PROJECT_NAME@

File: agent/src/fty_kpi_power_uptime.cc (hunk 2 above)
Comment (ericclappier-eaton) [reply agent/src/fty_kpi_power_uptime.cc]: Take into account

File: lib/src/dc.cc (hunk 5 above)
Comment (ericclappier-eaton) [reply lib/src/dc.cc]: Take into account

File: resources/fty-kpi-power-uptime.service.in (hunk 6 above)
Comment (ericclappier-eaton) [reply resources/fty-kpi-power-uptime.service.in]: Take into account

File: agent/CMakeLists.txt (hunk 1 above)
Comment (ericclappier-eaton) [reply agent/CMakeLists.txt]: Take into account

File: lib/CMakeLists.txt (hunk 3 above)
Comment (ericclappier-eaton) [reply lib/CMakeLists.txt]: Take into account

File: lib/CMakeLists.txt (hunk 4 above)
Comment (ericclappier-eaton) [reply lib/CMakeLists.txt:9]: Prefer the exhaustive list

**Pull request merged**